Pinta.Docking/DockLibrary/DockItem.cs

```diff
@@ -0,0 +1,607 @@
+
+				VBox box = new VBox ();
+				box.Show ();
+				box.PackStart (TitleTab, false, false, 0);
+				box.PackStart (Widget, true, true, 0);
+				floatingWindow.Add (box);
+				floatingWindow.DeleteEvent += delegate (object o, DeleteEventArgs a) {
+					if (behavior == DockItemBehavior.CantClose)
+						Status = DockItemStatus.Dockable;
+					else
+						Visible = false;
+					a.RetVal = true;
+				};
+			}
+			floatingWindow.Move (rect.X, rect.Y);
+			floatingWindow.Resize (rect.Width, rect.Height);
+			floatingWindow.Show ();
+			if (titleTab != null)
+				titleTab.UpdateBehavior ();
+			Widget.Show ();
+		}
+		
+		void ResetFloatMode ()
+		{
+			if (floatingWindow != null) {
+				// The widgets have already been removed from the window in ResetMode
+				floatingWindow.Dispose ();
+				floatingWindow = null;
+				if (titleTab != null)
+					titleTab.UpdateBehavior ();
+			}
+		}
+		
+		internal Gdk.Rectangle FloatingPosition {
+			get {
+				if (floatingWindow != null) {
+					int x,y,w,h;
+					floatingWindow.GetPosition (out x, out y);
+					floatingWindow.GetSize (out w, out h);
+					return new Gdk.Rectangle (x,y,w,h);
+				}
+				else
+					return Gdk.Rectangle.Zero;
+			}
+		}
+		
+		internal void ResetMode ()
+		{
+			if (Widget.Parent != null)
+				((Gtk.Container) Widget.Parent).Remove (Widget);
+			if (TitleTab.Parent != null)
+				((Gtk.Container) TitleTab.Parent).Remove (TitleTab);
+
+			ResetFloatMode ();
+			ResetBarUndockMode ();
+		}
+		
+		internal void SetAutoHideMode (Gtk.PositionType pos, int size)
+		{
+			ResetMode ();
+			if (widget != null) {
+				widget.Hide (); // Avoids size allocation warning
+				if (widget.Parent != null) {
+					((Gtk.Container)widget.Parent).Remove (widget);
+				}
+			}
+			dockBarItem = frame.BarDock (pos, this, size);
+			if (titleTab != null)
+				titleTab.UpdateBehavior ();
+
+			SetRegionStyle (frame.GetRegionStyleForItem (this));
+		}
+		
+		void ResetBarUndockMode ()
+		{
+			if (dockBarItem != null) {
+				dockBarItem.Close ();
+				dockBarItem = null;
+				if (titleTab != null)
+					titleTab.UpdateBehavior ();
+			}
+		}
+		
+		internal int AutoHideSize {
+			get {
+				if (dockBarItem != null)
+					return dockBarItem.Size;
+				else
+					return -1;
+			}
+		}
+
+		internal void Minimize ()
+		{
+			dockBarItem.Minimize ();
+		}
+
+		internal bool IsPositionMarker {
+			get {
+				return isPositionMarker;
+			}
+			set {
+				isPositionMarker = value;
+			}
+		}
+		
+		string GetWindowTitle ()
+		{
+			if (Label.IndexOf ('<') == -1)
+				return Label;
+			try {
+				XmlDocument doc = new XmlDocument ();
+				doc.LoadXml ("<a>" + Label + "</a>");
+				return doc.InnerText;
+			} catch {
+				return label;
+			}
+		}
+
+		internal bool ShowingContextMemu { get ; set; }
+		
+		internal void ShowDockPopupMenu (uint time)
+		{
+			Menu menu = new Menu ();
+			
+			// Hide menuitem
+			if ((Behavior & DockItemBehavior.CantClose) == 0) {
+				MenuItem mitem = new MenuItem (("Hide"));
```

Comment:
I think this were translatable strings in the `master` branch? (using `Catalog.GetString()`). In the GTK3 build you can use Pinta.Core.Translations, but we could move that elsewhere if needed for Pinta.Docking to depend on it.

Reply:
I referenced now Pinta.Core (for peaople that would want to use Pinta.Docking, it will not be so hard to remove it)

Reply:
Ok, sounds good!